umm-spec-lib/src/cmr/umm_spec/validation/spatial.clj

```diff
@@ -32,16 +42,29 @@
     (update-in spatial-extent path (fn [geometry] (map #(assoc % :coord-system coord-sys) geometry)))
     spatial-extent))
 
+(defn- validate-enum
+  "Validate if value exists in enums if value is not nil."
+  [field-path value enums]
+  (when value
+    (when-not (->enum value enums)
+      {field-path
+       [(format "Value (\"%s\") not found in enum (possible values: [%s])"
+                value
+                (->> (map #(format "\"%s\"" (csk/->SCREAMING_SNAKE_CASE_STRING %)) enums)
+                     sort ; Sorting to make order predictable for testing.
+                     (string/join ",")))]})))
+
 (defn- validate-spatial-representation
  "Validate that the granule spatial representation is valid if the spatial extent
  has geometries"
  [field-path spatial-extent]
  (let [granule-spatial-representation (:GranuleSpatialRepresentation spatial-extent)
        geometry (util/remove-nil-keys
                  (dissoc (get-in spatial-extent [:HorizontalSpatialDomain :Geometry]) :CoordinateSystem))]
-  (when (and (not (empty? geometry))
-             (nil? granule-spatial-representation))
-   {field-path ["Granule Spatial Representation must be supplied."]})))
+  (if (and (not (empty? geometry))
+           (nil? granule-spatial-representation))
+   {field-path ["Granule Spatial Representation must be supplied."]}
+   (validate-enum field-path granule-spatial-representation valid-granule-spatial-representations))))
```

Comment:
We only want to validate if the geometry is not empty.  But your code will validate if geometry is empty too. 
(when (not (empty? geometry))
    (if (nil? granule-spatial-representation)
      {field-path ["Granule Spatial Representation must be supplied."]}
     (validate-enum field-path granule-spatial-representation valid-granule-spatial-representations)))

Reply:
Can there not exist a case where the geometry is empty but the GranuleSpatialRepresentation is provided for ingest?

Reply:
That's a question for @eereiter. I was thinking about that too. But the doc string says: "Validate that the granule spatial representation is valid if the spatial extent has geometries", so either change the doc string, or change the code.

Reply:
I think validating the granule spatial representation anytime it is supplied is the correct thing to do. Cayvon explained that a nil granule spatial representation will pass the validation, so I think the code is good. Just change the docstring.